contracts/utils/OnlyStealthRelayer.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+
+interface IOnlyStealthRelayer {
+    event StealthRelayerSet(address _stealthRelayer);
+
+    function stealthRelayer() external view returns (address _stealthRelayer);
+
+    function setStealthRelayer(address _stealthRelayer) external;
+}
+
+/*
+ * OnlyStealthRelayerAbstract
+ */
+abstract contract OnlyStealthRelayer is IOnlyStealthRelayer {
+    address public override stealthRelayer;
+
+    constructor(address _stealthRelayer) public {
+        _setStealthRelayer(_stealthRelayer);
+    }
+
+    modifier onlyStealthRelayer() {
+        require(stealthRelayer == address(0) || msg.sender == stealthRelayer, "OnlyStealthRelayer::msg-sender-not-stealth-relayer");
```

Comment:
```suggestion
        require(stealthRelayer == address(0) || msg.sender == stealthRelayer, "OnlyStealthRelayer::not-realyer");
```
Ive done some research, and if require messages are bigger than 32 bytes they take 32 more bytes. So, in order to make things cheaper, we need to shorten our messages 😭

Reply:
nice catch!